pyqg/kernel.pyx

```diff
@@ -516,7 +516,7 @@ cdef class PseudoSpectralKernel:
     property ufull:
         def __get__(self):
             return np.asarray(self.u) + \
-                np.asarray(self.Ubg)[:,np.newaxis,np.newaxis]
+                np.expand_dims(np.asarray(self.Ubg),axis=2)
```

Comment:
Absolutely! This comes from needing to essentially needing to tuck a new dimension in the "middle" of the Ubg dimensions, since it can now have structure in both the j and k indexes. 

The bracketed notation using `np.newaxis` does not necessarily play well and give you the shape needed for u and Ubg to be broadcast together. However, `np.expand_dims` lets you arbitrarily choose where you want to pop in your new dimension, and coerce the data into the shape needed.